backend/routes/nft.go

```diff
@@ -195,3 +206,143 @@ func mintNFTDevnet(w http.ResponseWriter, r *http.Request) {
 	w.Header().Set("Access-Control-Allow-Origin", "*")
 	w.Write([]byte("Minted NFT on devnet"))
 }
+
+func LikeNFT(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Access-Control-Allow-Origin", "*")
+	w.Header().Set("Content-Type", "application/json")
+
+	if r.Method != http.MethodPost {
+		http.Error(w, "Method not allowed", http.StatusMethodNotAllowed)
+		return
+	}
+
+	var nftlikeReq NFTLikesRequest
+	err := json.NewDecoder(r.Body).Decode(&nftlikeReq)
+	if err != nil {
+		http.Error(w, errors.Wrap(err, "error decoding JSON").Error(), http.StatusBadRequest)
+		return
+	}
+
+	log.Printf("Received request: %+v", nftlikeReq)
+
+	// Check if the user has already liked the NFT
+	rows, err := core.ArtPeaceBackend.Databases.Postgres.Query(context.Background(), "SELECT * FROM NFTLikes WHERE nftKey = $1 AND liker = $2", nftlikeReq.NFTKey, nftlikeReq.UserAddress)
+	if err != nil {
+		http.Error(w, errors.Wrap(err, "error querying database").Error(), http.StatusInternalServerError)
+		return
+	}
+	defer rows.Close()
+
+	if !rows.Next() {
+		// No rows returned, user hasn't liked the NFT yet
+		_, err = core.ArtPeaceBackend.Databases.Postgres.Exec(context.Background(), "INSERT INTO NFTLikes (nftKey, liker) VALUES ($1, $2)", nftlikeReq.NFTKey, nftlikeReq.UserAddress)
+		if err != nil {
+			http.Error(w, errors.Wrap(err, "error inserting into database").Error(), http.StatusInternalServerError)
+			return
+		}
+		fmt.Println("NFT Liked By User")
+		w.WriteHeader(http.StatusOK)
+		w.Write([]byte("NFT Liked By User"))
+		return
+	}
+
+	// User has already liked the NFT
+	fmt.Println("NFT Already Liked By User")
+	w.WriteHeader(http.StatusOK)
+	w.Write([]byte("NFT Already Liked By User"))
+}
+
+func UnLikeNFT(w http.ResponseWriter, r *http.Request) {
+
+	w.Header().Set("Access-Control-Allow-Origin", "*")
+	w.Header().Set("Content-Type", "application/json")
+
+	if r.Method != http.MethodPost {
+		http.Error(w, "Method not allowed", http.StatusMethodNotAllowed)
+		return
+	}
+
+	var nftlikeReq NFTLikesRequest
+	err := json.NewDecoder(r.Body).Decode(&nftlikeReq)
+	if err != nil {
+		http.Error(w, errors.Wrap(err, "error decoding JSON").Error(), http.StatusBadRequest)
+		return
+	}
+
+	log.Printf("Received request: %+v", nftlikeReq)
+
+	// check if the user has like the nft
+	rows, err := core.ArtPeaceBackend.Databases.Postgres.Query(context.Background(), "SELECT * FROM NFTLikes WHERE nftKey = $1 AND liker = $2", nftlikeReq.NFTKey, nftlikeReq.UserAddress)
```

Comment:
This check isn't really needed, because the delete just wont do anything if there is no like

Reply:
This works fine